mining/src/mempool/validate_and_insert_transaction.rs

```diff
@@ -68,7 +68,16 @@ impl Mempool {
             }
         }
 
-        self.validate_transaction_in_context(&transaction)?;
+        // Calc the new mass including storage mass
+        let contextual_mass = consensus
+            .calculate_transaction_storage_mass(&transaction)
+            .and_then(|m| m.checked_add(transaction.calculated_compute_mass.unwrap()))
+            .ok_or(RuleError::RejectStorageMassIncomputable(transaction_id))?;
+
+        self.validate_transaction_in_context(&transaction, contextual_mass)?;
+
+        // Set the inner mass field post all validations
+        transaction.tx.set_mass(contextual_mass);
```

Comment:
This field is used inside `validate_mempool_transaction` so it should be calculated there or before